### ISSUE TITLE
Add more security to Apache

### ISSUE THREAD

**Issue by danielholm** [kyrofa/owncloud-snap#59]
Maybe ModSecurity[1] and/or fail2ban[2]?

Of course this would add more dependencies that has to be packaged with the snap, but I think it's very much worth it.

1: https://github.com/SpiderLabs/ModSecurity
2: https://www.techandme.se/fail2ban-owncloud/

**Comment by oparoz** [comment]
I agree, but the problem is that this snap is supposed to run on different types of hardware and some may not be able to run mod_security that well. Maybe we can include the module without loading it.

fail2ban would be difficult to implement without also including a firewall. An ownCloud app would probably work better for now.

**Comment by kyrofa** [comment]
Indeed, fail2ban is probably better suited to being its own snap, as it'll have a number of difficulties to overcome. Such a thing could be included in the owncloud-pi image though, once it exists.

**Comment by oparoz** [comment]
https://github.com/owncloud/pi-image/issues/37
